lib/internal/bootstrap/node.js

```diff
@@ -311,6 +302,32 @@
     }
   }
 
+  function setupTraceCategoryState() {
+    const { traceCategoryState } = internalBinding('trace_events');
+    const kCategoryAsyncHooks = 0;
+    let traceEventsAsyncHook;
+
+    function toggleTraceCategoryState() {
+      // Dynamically enable/disable the traceEventsAsyncHook
+      const asyncHooksEnabled = !!traceCategoryState[kCategoryAsyncHooks];
+
+      if (asyncHooksEnabled) {
+        // Lazy load internal/trace_events_async_hooks only if the async_hooks
+        // trace event category is enabled.
+        if (!traceEventsAsyncHook) {
+          traceEventsAsyncHook =
+            NativeModule.require('internal/trace_events_async_hooks');
+        }
+        traceEventsAsyncHook.enable();
+      } else if (!asyncHooksEnabled && traceEventsAsyncHook) {
```

Comment:
is the `!asyncHooksEnabled &&` necessary in the condition here?

Reply:
Yes, we only want to turn it off if that specific category is turned off

Reply:
I might be missing something, but isn't `asyncHooksEnabled` necessarily false on the `else` side of branch?